### DISCUSSION TITLE
How to transform StreamGeometry?

### DISCUSSION THREAD

**Discussion by shin0kaze** [AvaloniaUI/Avalonia#17412] [Q&A] [answered]
```axaml
<StreamGeometry>
  <StreamGeometry.Transform>
    <ScaleTransform ScaleX="2" ScaleY="3" />
  </StreamGeometry.Transform>
  <!--Unable to find suitable setter or adder for property Child of type Avalonia.Controls:Avalonia.Controls.Decorator for argument Avalonia.Base:Avalonia.Media.StreamGeometry, available setter parameter lists are:--> 
</StreamGeometry>
```
```axaml
<StreamGeometry Transform="{DynamicResource testTransform}">
  m 0,0 5,5
  <!--'No Content property or any Add methods found for type Avalonia.Base:Avalonia.Media.StreamGeometry.-->
</StreamGeometry>
```
None of above and many others of my attempts works.
So how can I adjust it scale, position to use in GroupGeometry and/or CombinedGeometry?

**Comment by shin0kaze** [comment]
Hmm, I can create a nested GeometryGroup with the desired transform that contains my StreamGeometry.
If I define them in resources, will they apply transform each time when I use them, or just once? Is it fine to use many GeometryGroups?

**Reply by LaurentInSeattle** [reply, marked as answer]
I would leave the geometry alone... 
And instead use a View box or apply a render transform to the UI object that uses that geometry.

HTH

**Reply by shin0kaze** [reply]
Yes. I use RenderTransform. The thing is that StreamGeometry does not support transforms directly, so I wrap it in GeometryGroup that supports transforms.
